staging/src/k8s.io/apiserver/pkg/storage/etcd3/api_object_versioner.go

```diff
@@ -94,11 +95,11 @@ func (a APIObjectVersioner) ParseResourceVersion(resourceVersion string) (uint64
 	}
 	version, err := strconv.ParseUint(resourceVersion, 10, 64)
 	if err != nil {
-		return 0, storage.NewInvalidError(field.ErrorList{
+		return 0, errors.NewBadRequest(
```

Comment:
this looks good to me, @deads2k is there any historical reason NewBadRequest was not used initially before?

Reply:
As this method is added by Daniel and approved by Jordon,
/cc liggitt lavalamp

Reply:
this utility method returning an API error is a little weird... normally, the thing calling this method in an API context would wrap in an API error... it looks like we do that in ~1/3 of the places we call this